pkg/local_object_storage/metabase/control.go

```diff
@@ -94,11 +94,18 @@ func (db *DB) Reset() error {
 	return db.init(true)
 }
 
+// asyncErrHandlingCapacity defines max async errors number
+// that can be handled in parallel.
+const asyncErrHandlingCapacity = 1000
+
 func (db *DB) init(reset bool) error {
 	if db.mode.NoMetabase() || db.mode.ReadOnly() {
 		return nil
 	}
 
+	db.errChan = make(chan error, asyncErrHandlingCapacity)
```

Comment:
But why are you complicating it this much? In normal case we expect exactly zero errors, so this code is not supposed to be running at all, let alone with concurrency. If it wastes some microseconds for logging it's not critical.

Reply:
i do not want to spend this time in the bolt TX

> so this code is not supposed to be running at all

but if it run, it is supposed to be run in cycle until we fix it or drop problem object

Reply:
so you are for just logging?

Reply:
> but if it run, it is supposed to be run in cycle until we fix it or drop problem object

This cycle starts once in a while and ends pretty quickly (depending on the DB).

> so you are for just logging?

I think it's sufficient.

Reply:
cant say i fully agree, IMO this code is not complex at all, and i would never log anything in a db transaction, but let it be your way